Thank you for testing this and for the feedback!  What's going on? Hmm..

Did aom build previously? Yes, it appears so: https://hydra.nixos.org/job/nixpkgs/trunk/libaom.x86_64-darwin (for example: https://hydra.nixos.org/build/86083792/log).

If I'm understanding this right, Borg did try building latest on Darwin... and libaom built successfully, with the failure occurring [during the ffmpeg build](https://logs.nix.ci/?key=nixos/nixpkgs.52444&attempt_id=a001dda2-9bfa-4fea-9fda-042c08815e16):
```
[...]
LD      libavdevice/libavdevice.58.dylib
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_NSDictionary", referenced from:
      objc-class-ref in avfoundation.o
  "_OBJC_EHTYPE_$_NSException", referenced from:
      GCC_except_table4 in avfoundation.o
ld: symbol(s) not found for architecture x86_64
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [ffbuild/library.mak:103: libavdevice/libavdevice.58.dylib] Error 1
builder for '/nix/store/4633kfnwrbqfb6j0xd5hnj1m40g4kjm0-ffmpeg-full-4.1.drv' failed with exit code 2
error: build of '/nix/store/4633kfnwrbqfb6j0xd5hnj1m40g4kjm0-ffmpeg-full-4.1.drv' failed
```

And yes indeed ffmpeg_4 currently ("previously") builds on Darwin: https://hydra.nixos.org/job/nixpkgs/trunk/ffmpeg_4.x86_64-darwin (example: https://hydra.nixos.org/build/86416146).  This is expected but just checking :).

@Infinisil are you seeing the same behavior/error as Borg? And if it's not too much trouble: was the last commit (https://github.com/NixOS/nixpkgs/pull/52444/commits/d349cfee88384b4cc353dba8d1154f230520bcc4) necessary to fix libaom for you (if indeed it now builds again)?

--- a/pkgs/development/libraries/ffmpeg-full/default.nix
+++ b/pkgs/development/libraries/ffmpeg-full/default.nix
@@ -140,7 +140,7 @@
  *  Darwin frameworks
  */
 , Cocoa, CoreAudio, CoreServices, AVFoundation, MediaToolbox
-, VideoDecodeAcceleration, CF
+, VideoDecodeAcceleration, cf-private
 }:
 
 /* Maintainer notes:
@@ -434,7 +434,7 @@ stdenv.mkDerivation rec {
     FILES+=($(ls $out/lib/*.dylib))
     for f in ''${FILES[@]}; do
       if [ ! -h "$f" ]; then
-        install_name_tool -change ${CF}/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation "$f"
+        install_name_tool -change ${cf-private}/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation "$f"
       fi
     done
   '';

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -9731,7 +9731,7 @@ in
     vid-stab = if stdenv.isDarwin then null else vid-stab;
     x265 = if stdenv.isDarwin then null else x265;
     xavs = if stdenv.isDarwin then null else xavs;
-    inherit (darwin) CF;
+    inherit (darwin) cf-private;
     inherit (darwin.apple_sdk.frameworks)
       Cocoa CoreServices CoreAudio AVFoundation MediaToolbox
       VideoDecodeAcceleration;

